src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -556,13 +556,14 @@ private BoundExpression MakeImplicitConversion(BoundExpression rewrittenOperand,
 
                 TypeSymbol userDefinedConversionRewrittenType = conversion.Method.ReturnType;
 
-                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0])
+                // the conversion only needs to happen for non-nullable valuetypes
+                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0] && !userDefinedConversionRewrittenType.CanBeAssignedNull())
                 {
                     Debug.Assert(rewrittenOperand.Type.IsNullableType());
                     Debug.Assert(rewrittenOperand.Type.GetNullableUnderlyingType() == conversion.Method.ParameterTypes[0]);
-                    Debug.Assert(!userDefinedConversionRewrittenType.IsNullableType());
 
                     // Lifted conversion, wrap return type in Nullable
+                    // BUG: Construct(...) should check the conversion is correct, as it is now, it will not check generic type parameter constraints
```

Comment:
When there is ```Nullable<string>``` in code, we should be able to construct the symbol so that constraints could be checked later.